### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by brohamgoham**
Attempting to imporve error handling and make it more consistent across codebase

### CODE REVIEW

**Comment by brohamgoham** [comment]
#50

**Comment by brohamgoham** [comment]
@Vid201 hey can you please see if im going in the right direction, also should i move tonic::Status errors into a errors.rs file in gprc dir or bundler dir.

Bear with me as Im still trying to understand the whole repo, thank you

**Comment by Vid201** [comment]
> @Vid201 hey can you please see if im going in the right direction, also should i move tonic::Status errors into a errors.rs file in gprc dir or bundler dir.
> 
> Bear with me as Im still trying to understand the whole repo, thank you

It looks good at the moment. It would be perfect if all tonic errors would go to grpc (since they are specific for grpc protocol).

**Comment by brohamgoham** [comment]
> > @Vid201 hey can you please see if im going in the right direction, also should i move tonic::Status errors into a errors.rs file in gprc dir or bundler dir.
> > Bear with me as Im still trying to understand the whole repo, thank you
> 
> It looks good at the moment. It would be perfect if all tonic errors would go to grpc (since they are specific for grpc protocol).
> 
> Make sure to make PR draft until it's ready to be reviewed.

yes i was going to add tonic errors in the next commit,
Thanks

File: crates/grpc/src/uopool.rs [hunk 1]
```diff
@@ -27,6 +20,12 @@ use ethers::{
     providers::{Http, Middleware, Provider},
     types::{Address, H256, U256, U64},
 };
+use std::{
```
Comment (brohamgoham) [inline crates/grpc/src/uopool.rs:23]: running cargo fmt did this

**Comment by brohamgoham** [comment]
had an issue on the aync functions that used `#[async_trait]` in grpc/uopool.rs when I tried to use GrpcErrors instead of tonic::Status 

error i got `change the output type to match the trait: `Pin<Box<(dyn Future<Output = std::result::Result<tonic::Response<proto::bundler::SetModeResponse>, Status>> + Send + 'async_trait)>>`rustcE0053`

anyways if you could review and point me in right direction that would be greatly appreciated

**Comment by Vid201** [comment]
> had an issue on the aync functions that used `#[async_trait]` in grpc/uopool.rs when I tried to use GrpcErrors instead of tonic::Status
> 
> error i got `change the output type to match the trait: Pin<Box<(dyn Future<Output = std::result::Result<tonic::Response<proto::bundler::SetModeResponse>, Status>> + Send + async_trait)>>`rustcE0053`
> 
> anyways if you could review and point me in right direction that would be greatly appreciated

Hey @brohamgoham . This looks good at the moment, kudos.

If you want, there are several things that would be very helpful:
- crate `rpc` should contains error definitions that are returned by JSON-RPC API (text messages, error codes ...)
- crate 'grpc' contains errors related to the gRPC communication
- crates `bundler` and `uopool` contains errors specific to bundler and uopool
- crate 'contract' contains errors specific to calling entry point smart contract

**Comment by Vid201** [comment]
Hey @brohamgoham , are you still working on that? Just letting you know I refactored some of the code (https://github.com/Vid201/aa-bundler/commit/4974f2dbf497cb240335705ab19af43f07593f24) which affects your work on error handling.

**Comment by Vid201** [comment]
> Hey @brohamgoham , are you still working on that? Just letting you know I refactored some of the code ([4974f2d](https://github.com/Vid201/aa-bundler/commit/4974f2dbf497cb240335705ab19af43f07593f24)) which affects your work on error handling.

@brohamgoham I am closing this PR since there is no activity. Can be reopened in the future.

**Pull request closed without merging**